### PR TITLE
Add Electron Fiddle

### PR DESCRIPTION
**Pull request by Crilum**
[Electron Fiddle](https://www.electronjs.org/fiddle) lets you create and play with small Electron experiments. It greets you with a quick-start template after opening – change a few things, choose the version of Electron you want to run it with, and play around. Then, save your Fiddle either as a GitHub Gist or to a local folder. Once pushed to GitHub, anyone can quickly try your Fiddle out by just entering it in the address bar.

### CODE REVIEW

**Comment by Itai-Nelken** [comment]
@Crilum can you try to remove the extra commits in your next pr?

**Comment by cycool29** [comment]
Ya, learn to use git. This should be [easily done]( https://stackoverflow.com/questions/25101983/can-i-revert-commits-directly-on-github) with `git reset`. 
I think there is also a function to revert changes in Github Desktop, not very sure about it.

**Comment by Crilum** [comment]
@Itai-Nelken @cycool29 Yeah.. Sorry. Thanks for telling me about `git reset`.

**Comment by Crilum** [comment]
@cycool29 It looks like `git reset` is kind of dangerous.. Would `git revert` be a better choice? Also, if I were to do one of these commands, would it get rid of all the commits in the PR? Or would it just make a new commit reverting all the other ones?

**Comment by Crilum** [comment]
I think I might close all my PRs, delete my fork, create another one, upload my apps ***correctly***, and make new PRs for the same apps. Do you guys think this is a good idea? As you can see with every PR, my fork is a complete mess...

**Comment by Itai-Nelken** [comment]
@Crilum wait until all your pr's are merged/closed and then delete your fork and reclone. we can squash and merge the pr's so they appear as a single commit.

**Comment by Crilum** [comment]
Yeah, that's the other thing I was thinking too. Does this PR look good?

**Comment by Crilum** [comment]
So, I think this is affected by the same problem as Deskreen. But running it with --no-sandbox gives me this:
```
$ electron-fiddle --no-sandbox
Creating main window
Usage: main [options] [command]

Options:
  -h, --help                                   display help for command

Commands:
  bisect [options] <goodVersion> <badVersion>  Find where regressions were introduced
  test [options]                               Test a fiddle
  help [command]                               display help for command


Example calls:
  $ electron-fiddle bisect 10.0.0 11.2.0 --fiddle /path/to/fiddle
  $ electron-fiddle bisect 10.0.0 11.2.0 --fiddle /path/to/fiddle --full
  $ electron-fiddle test --version 11.2.0 --fiddle /path/to/fiddle
  $ electron-fiddle test --version 11.2.0 --fiddle 8c5fc0c6a5153d49b5a4a56d3ed9da8f
  $ electron-fiddle test --version 11.2.0 --fiddle https://gist.github.com/ckerr/8c5fc0c6a5153d49b5a4a56d3ed9da8f/

CommanderError: (outputHelp)
    at Command._exit (/usr/lib/electron-fiddle/resources/app.asar/node_modules/commander/index.js:888:26)
    at Command.help (/usr/lib/electron-fiddle/resources/app.asar/node_modules/commander/index.js:2050:10)
    at Command._parseCommand (/usr/lib/electron-fiddle/resources/app.asar/node_modules/commander/index.js:1521:14)
    at Command.parse (/usr/lib/electron-fiddle/resources/app.asar/node_modules/commander/index.js:1292:10)
    at Command.parseAsync (/usr/lib/electron-fiddle/resources/app.asar/node_modules/commander/index.js:1318:10)
    at Object.m [as processCommandLine] (/usr/lib/electron-fiddle/resources/app.asar/dist/src/main/main.js:32:3637)
    at q (/usr/lib/electron-fiddle/resources/app.asar/dist/src/main/main.js:44:997) {
  code: 'commander.help',
  exitCode: 1,
  nestedError: undefined
}
Electron Fiddle is exiting with code 2
```

I think Fiddle does not have a --no-sandbox flag, but electron does and that is why it displays a help?

Don't know about the error...

**Comment by Crilum** [comment]
This is the issue I open in the fiddle repository, for anyone interested:
https://github.com/electron/fiddle/issues/900#issue-1052792250

**Comment by ryanfortner** [comment]
@Crilum your script can be updated to v0.27.0

File: apps/Electron Fiddle/install-32
Comment (cycool29) [inline apps/Electron Fiddle/install-32:3]:
```suggestion
```
This is unneeded now.

File: apps/Electron Fiddle/install-32
Comment (cycool29) [inline apps/Electron Fiddle/install-32:9]:
```suggestion
```

File: apps/Electron Fiddle/install-32
Comment (cycool29) [inline apps/Electron Fiddle/install-32:10]: Update to 0.27.2.
```suggestion
install_packages https://github.com/electron/fiddle/releases/download/v0.27.2/electron-fiddle_0.27.2_armhf.deb
```

File: apps/Electron Fiddle/install-32
Comment (cycool29) [inline apps/Electron Fiddle/install-32:11]:
```suggestion
rm -rf ${fiddle_ver} || sudo rm -rf ${fiddle_ver}
```
`install_packages` already handles that.

File: apps/Electron Fiddle/install-64
Comment (cycool29) [inline apps/Electron Fiddle/install-64:3]:
```suggestion
```

File: apps/Electron Fiddle/install-32
Comment (cycool29) [inline apps/Electron Fiddle/install-32:8]:
```suggestion
```

`error` is already defined in `api`.

File: apps/Electron Fiddle/install-64
Comment (cycool29) [inline apps/Electron Fiddle/install-64:8]:
```suggestion
```

File: apps/Electron Fiddle/install-64
Comment (cycool29) [inline apps/Electron Fiddle/install-64:10]:
```suggestion
```

File: apps/Electron Fiddle/install-64
Comment (cycool29) [inline apps/Electron Fiddle/install-64:11]:
```suggestion
install_packages https://github.com/electron/fiddle/releases/download/v0.27.2/electron-fiddle_0.27.2_arm64.deb
```

File: apps/Electron Fiddle/install-64
Comment (cycool29) [inline apps/Electron Fiddle/install-64:12]:
```suggestion
```

File: apps/Electron Fiddle/uninstall
Comment (cycool29) [inline apps/Electron Fiddle/uninstall:3]:
```suggestion

```

File: apps/Electron Fiddle/uninstall
Comment (cycool29) [inline apps/Electron Fiddle/uninstall:8]:
```suggestion
```

File: apps/Electron Fiddle/uninstall
Comment (cycool29) [inline apps/Electron Fiddle/uninstall:10]:
```suggestion
purge_packages | exit 1
```

**Review by cycool29 — CHANGES_REQUESTED** [review]
Few improvements.

**Comment by Crilum** [comment]
Yeah, thanks @cycool29. But, Fiddle is an Electron app, and it needs the `--no-seccomp-sandbox` flag, or whatever it was, to work, but, Fiddle has it's own terminal interface, and it blocks commands from getting to Electron, so this can't be added until electron has fixed that. I have made an issue, but there isn't a lot of progress, as far as I can tell.

**Comment by Botspot** [comment]
> Yeah, thanks @cycool29. But, Fiddle is an Electron app, and it needs the `--no-seccomp-sandbox` flag, or whatever it was, to work, but, Fiddle has it's own terminal interface, and it blocks commands from getting to Electron, so this can't be added until electron has fixed that. I have made an issue, but there isn't a lot of progress, as far as I can tell.

I encountered this same problem with the Downgrade Chromium app on Bullseye. To work around it, I had to [add](https://github.com/Botspot/pi-apps/commit/7c470851010338b84d3ec8f994f6ece83b39f2c5#diff-91c8a44db067d1d87a2ad13c766ae68e3e5ce0251805813b535b0fc86995a7e0R145) the `--no-seccomp-sandbox` flag to all chromium versions under 92. But the latest Chromium on Bullseye doesn't need that flag. Also, Scratch 3 is an electron app made by RPi developers. How did they get past the `--no-seccomp-sandbox` flag?

**Comment by Crilum** [comment]
> > Yeah, thanks @cycool29. But, Fiddle is an Electron app, and it needs the `--no-seccomp-sandbox` flag, or whatever it was, to work, but, Fiddle has it's own terminal interface, and it blocks commands from getting to Electron, so this can't be added until electron has fixed that. I have made an issue, but there isn't a lot of progress, as far as I can tell.
> 
> I encountered this same problem with the Downgrade Chromium app on Bullseye. To work around it, I had to [add](https://github.com/Botspot/pi-apps/commit/7c470851010338b84d3ec8f994f6ece83b39f2c5#diff-91c8a44db067d1d87a2ad13c766ae68e3e5ce0251805813b535b0fc86995a7e0R145) the `--no-seccomp-sandbox` flag to all chromium versions under 92. But the latest Chromium on Bullseye doesn't need that flag. Also, Scratch 3 is an electron app made by RPi developers. How did they get past the `--no-seccomp-sandbox` flag?

I don't know, but Electron uses chromium 7 or 8 (I think), so..

File: apps/Electron Fiddle/install-32
Comment (Crilum) [reply apps/Electron Fiddle/install-32:8]: Why? This just makes updating Fiddle easier.

File: apps/Electron Fiddle/install-32
Comment (Crilum) [reply apps/Electron Fiddle/install-32:10]: oh, ok.

File: apps/Electron Fiddle/install-32
Comment (cycool29) [reply apps/Electron Fiddle/install-32:4]: The variable is only used one time in the script. We can just simply change the version number at 
`install_packages https://github.com/electron/fiddle/releases/download/v0.27.2/electron-fiddle_0.27.2_armhf.deb || exit 1`.

So I don't think defining a variable is needed.

**Comment by Crilum** [comment]
I fixed the conflicts, so this would be ready for merging, but the sandbox stuff hasn't been resolved..

**Comment by Crilum** [comment]
@ericzhao fixed the problem, we just need a build, or a release...

**Comment by erickzhao** [comment]
We just finished doing reviews for a bunch of PRs we had in the backlog. I'll try to get a release in for this week. :)

**Comment by Crilum** [comment]
Fiddle has fixed the problem in their latest release (Thanks @erickzhao!), I haven't tested this yet, but should be good to go!

**Comment by Crilum** [comment]
Ok, I've just got around to actually testing this, and it wasn't installing because it thought the `version` variable was longer then it actually was. Fixed that, and now it works like a charm! I don't even need to use `--no-seccomp-sandbox`.

I think this is ready for merge, @Botspot.

**Comment by ryanfortner** [comment]
lgtm

**Comment by ryanfortner** [comment]
@Botspot

**Comment by theofficialgman** [comment]
seems like a fun tool. unfortunately it also breaks on bionic/buster due to the electron 18 bug https://github.com/electron/electron/issues/33533, since it downloads the latest release for the "run" button.

**Comment by Crilum** [comment]
Darn.. This has been sitting here for like 5 months... oh well. I guess we'll have to wait for electron. :man_shrugging:

**Comment by theofficialgman** [comment]
ok give this PR just a little while longer, 18.2.4 isn't showing up in-app yet, it probably is just a cache somehwere, just give it a few hours or maybe till tomorrow

**Comment by theofficialgman** [comment]
alright its there now. @Botspot , I'm merging, please run the clicklist updater

**Comment by Crilum** [comment]
Thanks @theofficialgman!

**Comment by Botspot** [comment]
Bitly links were created 15 hours ago.

**Pull request merged**
